src/include/duckdb/function/scalar_function.hpp

```diff
@@ -26,19 +29,22 @@ typedef unique_ptr<FunctionData> (*bind_scalar_function_t)(BoundFunctionExpressi
 //! Adds the dependencies of this BoundFunctionExpression to the set of dependencies
 typedef void (*dependency_function_t)(BoundFunctionExpression &expr, unordered_set<CatalogEntry *> &dependencies);
 
+//! The type of UDF functions
+typedef std::function<void(DataChunk &, ExpressionState &, Vector &)> udf_function_t;
```

Comment:
It looks like this is the same as `scalar_function_t`.

Reply:
This is a little problematic, I tried hard to use the scalar_function_t, but as it is a function pointer basically, it cannot  preserve states of the capture values from lambda functions:

```
auto function = [=] (DataChunk &input, ExpressionState &state, Vector &result) {
				UnaryExecutor::Execute<TA, TR>(input.data[0],
								result,
								input.size(),
								udf_func);
                             };
```
Such lambda uses the capture clause to get the address of the UDF function pointer (*udf_func), although the code compiles, it doesn't work properly at runtime (the values passed to the UDF function are very random). This is the reason why I had to define the new type udf_function_t because the std::function preserves the state of the capture values.

Reply:
How about changing scalar_function_t to std::function then? That should work, no?